src/persistence/settings.cpp

```diff
@@ -2318,12 +2331,13 @@ bool Settings::getAutoLogin() const
     return autoLogin;
 }
 
-void Settings::setAutoLogin(bool state)
+void Settings::onSetAutoLogin(bool state)
```

Comment:
This was changed to update and read changes through signals rather than direct object access. Now it should emit a signal when the checkbox is clicked which prompts settings to update the value internally. When the value is updated in Settings, another signal is sent back to the LoginScreen notifying it that the autologin value has changed, which updates the visual of the checkbox. This means that LoginScreen does not have to know anything about Settings to function, which makes the code more concise and modular.

A somewhat unlikely utility of this would be if the autologin property was changed elsewhere. The previous implementation would have no way of knowing a change had taken place, while this one would send a signal indicating the update. The usefulness of this implementation is much more obvious in the case of multiple windows representing the same group chat, for example.

Reply:
Again. You misunderstood my point. I'm not against using a signal/slot mechanism to interact between classes. I talk about method name. I think `Settings` shouldn't know is its methods used as an endpoint of the connection or not. The method should be `onXXX` only if they connected inside the class itself.
I.e.
```
void MyClass::init()
{
    connect(view, &MyView::autoLoginChanged, this, &MyClass::onAutoLoginChanged);
}
```
but
```
void MyClass::init()
{
    connect(view, &MyView::autoLoginChanged, service, &MyService::setAutoLogin);
}
```

Reply:
Yes, I see what you mean now. Sorry this took so long to respond to, I didn't realize Github would hide unresolved discussions when condensing multiple. Changed as indicated.

Reply:
I renamed onCurrentProfileChanged as well to match this line of thinking